src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -545,6 +545,54 @@ internal string GetUIVisualizerName(string serviceId, int id)
 
         private delegate IVariableInformation Traverse(IVariableInformation node);
 
+        // new function here
+        private string GetDisplayNameFromArrayIndex(uint arrayIndex, int rank, uint[] dimensions, bool isForward)
+        {
+            StringBuilder displayName = new StringBuilder();
+            // displayName.Append('[');
+
+            uint index = arrayIndex;
+
+            int i = rank - 1;
+            int inc = -1;
+            int endLoop = -1;
+
+            if (!isForward)
+            {
+                i = 0;
+                inc = 1;
+                endLoop = rank;
+            }
+
+            uint[] indices = new uint[rank];
+
+            while (i != endLoop)
+            {
+                uint dimensionSize = dimensions[i];
+                uint divResult = index / dimensionSize;
+                uint modResult = index % dimensionSize;
+
+                indices[i] = modResult;
+                index = divResult;
+
+                i += inc;
+            }
+
+            if (rank != 0)
+            {
+                displayName.Append(indices[0]);
+                for (i = 1; i < rank; i++)
+                {
+                    displayName.Append(',');
```

Comment:
You will also want to account for the current engine radix which you can get with `_engine.CurrentRadix()`.

You can use `StringBuilder.AppendFormat` and provide the correct format for decimal or hex.

Reply:
This is what I have tried, but the strings are not displaying correctly. I have tried looking at examples of `AppendFormat` as well...

```
string format = _process.Engine.CurrentRadix() == 16 ? "0x%08x" : "%d";
displayName.AppendFormat(CultureInfo.InvariantCulture, format, indices[0]);
```

Reply:
Have you tried example from https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.appendformat?view=net-6.0?